### DISCUSSION TITLE
Can custom CSS be integrated into the current login UI?

### DISCUSSION THREAD

**Discussion by cocadepoma** [zitadel/zitadel#7177] [Q&A] [answered]
Greetings ZITADEL community,

I'm exploring the feasibility of incorporating custom CSS into the current login UI to enhance the visual coherence with my company's brand. This potential feature could offer efficiency and flexibility. Any insights, documentation, or recommended practices for achieving this would be greatly appreciated.

Thank you.

**Comment by hifabienne** [comment]
Hei @soyl3y3nd4 
At the moment it is not possible to include custom CSS.
You can change the branding of the ui, which includes colours, font, and logo.
If you do want to fully customize the login, you do have the possibility to use our session API and implement it yourself.
You can find a guide on that here: https://zitadel.com/docs/guides/integrate/login-ui
We are also working on a sdk to make it easier. (its not yet finished) https://github.com/zitadel/typescript

**Reply by cocadepoma** [reply]
Thank you for your swift response!

I find the idea of incorporating custom CSS, similar to Keycloak, quite compelling. This functionality would empower users to modify specific elements like the background image, the user/password form, or even button animations without the need to construct an entire UI with OIDC flows.

I am curious if there are any plans to implement this feature in the future. I would be enthusiastic about contributing by actively seeking solutions or even implementing the feature.

**Reply by hifabienne** [reply, marked as answer]
We talked about that earlier on, but so far always decided against it as it opens up a security thread vector.
As soon as we allow customers to include custom css/code, we can't be sure that there is nothing that breaks the security.

We have an open issue, about adding more pre defined customization to the existing configurations: https://github.com/zitadel/zitadel/issues/3987
But it is also not yet planned as we want to focus on finishing the sdks first.

**Reply by cocadepoma** [reply]
Ok, I got it, thank you for your answer.
